### DISCUSSION TITLE
Project Ideas

### DISCUSSION THREAD

**Discussion by AlannaBurke** [uselagoon/lagoon#3691] [Google Season of Docs 2024]
We're looking at submitting to Google Season of Docs 2024 - let's brainstorm some possible projects.

**Comment by AlannaBurke** [comment]
## Technical diagram overhaul of Lagoon (and its components)

### Problem
Our technical diagrams are out of date, not all in one place, and difficult to manage and keep up to date.

https://lucid.app/lucidchart/cb441054-e04a-4389-b98b-c75bcda8ea0d/view?referringApp=slack&page=0_0#

Use structurizr? https://github.com/uselagoon/diagram-structurizr

If we’re unsure about structurizr, we could propose that the first part of the task is evaluating structurizr and other potential solutions, present a project plan, and establish which works best for us.

**Comment by AlannaBurke** [comment]
## Gather environment variables

### Problem
We have environment variables scattered throughout various parts of Lagoon. They are not fully documented, and users likely do not know about them and are unable to make use of them. 

### Hypothesis
If our environment variables were fully documented, our users could make use of them, enhancing their experience. 
Propose a solution

We’d like to comb through Lagoon and find all of the existing environment variables and document them, and future-proof a solution to ensure any future rogue variables are discovered and documented.

**Comment by AlannaBurke** [comment]
## Generate code examples in docs

### Problem
Our docs contain many code samples, and it’s difficult to keep them up to date - whether it’s because we don’t know they exist, or because the person updating the docs doesn’t know there’s been an update to the code. 

### Hypothesis
We should be able to generate code snippets from actual code and include these into the docs, making them update automatically if the code is updated.

**Comment by AlannaBurke** [comment]
## Manage image versions in examples (variable replacement)

### Problem
As our examples often include many Docker images, it is difficult to maintain a wide variety of examples and keep them all up to date with the latest versions of their respective included images.

**Comment by AlannaBurke** [comment]
## Sort docs into personas

### Problem
Currently our docs are organized by topic, but we have varied personas who use the project, and it would be a better experience for them to have documentation organized by persona.

**Comment by AlannaBurke** [comment]
## Can we write our docs to be optimized to be consumable by LLMs?

### Problem
Alanna writes like a human being

**Comment by yashpandey06** [comment]
Sounds interesting!

**Comment by Ayush9026** [comment]
Hi @AlannaBurke mam , impressive range of ideas covering diverse aspects! 👍

**Comment by AlannaBurke** [comment]
Update - our project was not selected, so i'll be closing this discussion and we'll try again net year!
